BLSQ/iaso
# Polio 687 rounds on calendar map

Since we can have different scopes for each campaign round, we need the polio calendar map to be able to switch which round is displayed

## Self proof reading checklist

- [NA] My migrations file are included
- [] Are there enough tests - this is polio, so no

## Changes

- Added possibility to choose which shapes to display: all (all rounds mixed), latest (current round, or last round that ended), or choose by round number. When choosing by round number, only the exact selection will be shown (eg: a campaign with 2 rounds will show no shape if you choose round 3):
    -  Added hooks and utils to make shapes and merged shapes sortable
    -  Added `MapRoundButton` and `MapRoundSelector`components
- Converted `CalendarMap`to TypeScript and added typing where needed
- Set `cacheTime` to `Infinity` in `useGetCampaigns` hook to avoid reloading all campaigns and shapes when switching language. The cache is still invalidated when creating or editing a campaign.
- Enabled passing an empty string as `labelString` to `InputComponent` without it falling back to using `formatMessage`


## How to test

- Make sure you have "multi round" campaigns in your DB. If not create at least one
- Go to the polio calendar and play with the dates, until you see your multi round campaign
- Play with the round selection button on the map

## Print screen / video


https://user-images.githubusercontent.com/38907762/203842414-fc99cf42-3d9e-4971-8589-98b073be9656.mov

## Note 

Needs to be merged after https://github.com/BLSQ/iaso/pull/188
